internal/exec/util/file.go

```diff
@@ -210,49 +214,59 @@ func (u Util) PerformFetch(f *FetchOp) error {
 	return nil
 }
 
-func (u Util) GetUserGroupID(l *log.Logger, user types.NodeUser, group types.NodeGroup) (*int, *int) {
-	if user.Name != "" {
-		usr, err := u.userLookup(user.Name)
-		if err != nil {
-			l.Crit("No such user %q: %v", user.Name, err)
-			return nil, nil
-		}
-		uid, err := strconv.ParseInt(usr.Uid, 0, 0)
-		if err != nil {
-			l.Crit("Couldn't parse uid %q: %v", usr.Uid, err)
-			return nil, nil
+// MkdirForFile helper creates the directory components of path.
+func MkdirForFile(path string) error {
+	return os.MkdirAll(filepath.Dir(path), DefaultDirectoryPermissions)
+}
+
+func (u Util) ResolveNodeUidAndGid(n types.Node, defaultUid, defaultGid int) (int, int, error) {
```

Comment:
Comment here saying what this does would be nice. The old one didn't have one, but it should have.